src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueueDebugView.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Collections.Generic
+{
+    internal sealed class PriorityQueueDebugView<TElement, TPriority>
+    {
+        private readonly PriorityQueue<TElement, TPriority> _queue;
+        private readonly bool _sort;
+
+        public PriorityQueueDebugView(PriorityQueue<TElement, TPriority> queue)
+        {
+            _queue = queue ?? throw new ArgumentNullException(nameof(queue));
+            _sort = true;
+        }
+
+        public PriorityQueueDebugView(PriorityQueue<TElement, TPriority>.UnorderedItemsCollection collection)
+        {
+            _queue = collection?._queue ?? throw new ArgumentNullException(nameof(collection));
+        }
+
+        [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+        public (TElement Element, TPriority Priority)[] Items
+        {
+            get
+            {
+                var list = new List<(TElement Element, TPriority Priority)>(_queue.UnorderedItems);
+                if (_sort)
+                {
+                    list.Sort((i1, i2) => _queue.Comparer.Compare(i1.Priority, i2.Priority));
```

Comment:
If no custom comparer is specified, a new comparer will be allocated every time `_queue.Comparer` is called:
```suggestion
                    IComparer<TPriority> comparer = _queue.Comparer;
                    list.Sort((i1, i2) => comparer.Compare(i1.Priority, i2.Priority));
```
Alternatively we could perhaps cache the comparer value returned by the queue property.

Reply:
> a new comparer will be allocated every time _queue.Comparer is called

It will? The property is implemented as:
```C#
=> _comparer ?? Comparer<TPriority>.Default
```
and Default is a singleton.

Reply:
> and Default is a singleton.

For some reason I assumed it wasn't. Looks good then.

Reply:
> For some reason I assumed it wasn't

That would be _really_ bad for all of our code paths as part of heapifying that do `Comparer<TPriority>.Default.Compare(...)` :smile: